src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,157 @@
+        {
+        }
+
+        protected override async Task<SyntaxNode?> GetInvocationExpressionSyntaxNodeAsync(CodeRefactoringContext context)
+        {
+            var syntaxNode = await context.TryGetRelevantNodeAsync<InvocationExpressionSyntax>().ConfigureAwait(false);
+            return syntaxNode;
+        }
+
+        private static bool ShouldStatementBeInlined(StatementSyntax statementSyntax)
+            => statementSyntax is ReturnStatementSyntax || statementSyntax is ExpressionStatementSyntax;
+
+        protected override bool IsMethodContainsOneStatement(SyntaxNode calleeMethodDeclarationSyntaxNode)
+        {
+            if (calleeMethodDeclarationSyntaxNode is MethodDeclarationSyntax declarationSyntax)
+            {
+                var blockSyntaxNode = declarationSyntax.Body;
+                // 1. If it is an ordinary method with block
+                if (blockSyntaxNode != null)
+                {
+                    var blockStatements = blockSyntaxNode.Statements;
+                    return blockStatements.Count == 1 && ShouldStatementBeInlined(blockStatements[0]);
+                }
+                else
+                {
+                    // 2. If it is an Arrow Expression
+                    var arrowExpressionNodes = declarationSyntax
+                        .DescendantNodes().Where(node => node.IsKind(SyntaxKind.ArrowExpressionClause)).ToImmutableArray();
+                    return arrowExpressionNodes.Length == 1;
+                }
+            }
+
+            return false;
+        }
+
+        protected override IParameterSymbol? GetParameterSymbol(SemanticModel semanticModel, SyntaxNode argumentSyntaxNode, CancellationToken cancellationToken)
+            => argumentSyntaxNode is ArgumentSyntax argumentSyntax
+                ? argumentSyntax.DetermineParameter(semanticModel, allowParams: true, cancellationToken)
+                : null;
+
+        protected override SyntaxNode GenerateLiteralExpression(ITypeSymbol typeSymbol, object? value)
+            => ExpressionGenerator.GenerateExpression(typeSymbol, value, canUseFieldReference: false);
+
+        protected override bool IsExpressionSyntax(SyntaxNode syntaxNode)
+            => syntaxNode is ExpressionSyntax;
+
+        protected override string GetIdentifierTokenTextFromIdentifierNameSyntax(SyntaxNode syntaxNode)
+        {
+            if (syntaxNode is IdentifierNameSyntax identifierNameSyntax)
+            {
+                return identifierNameSyntax.Identifier.ValueText;
+            }
+
+            return string.Empty;
+        }
+
+        protected override SyntaxNode GenerateArrayInitializerExpression(ImmutableArray<SyntaxNode> arguments)
+            => SyntaxFactory.InitializerExpression(SyntaxKind.ArrayInitializerExpression, SyntaxFactory.SeparatedList(arguments));
+
+        protected override SyntaxNode GenerateLocalDeclarationStatementWithRightHandExpression(
```

Comment:
i feel like SyntaxGenerator can do this.

Reply:
Yup, replace it with SyntaxGenerator.